tests/test_geoprocessing.py

```diff
@@ -1375,6 +1465,80 @@ def test_zonal_statistics_multipolygon(self):
                 'sum': 2
         }})
 
+    def test_zonal_statistics_3d_multipolygon(self):
+        """PGP.geoprocessing: test zonal stats function with 3d multipolygons."""
+        # create aggregating polygon
+        pixel_size = 30.0
+        n_pixels = 9
+
+        driver = ogr.GetDriverByName('GeoJSON')
+        aggregating_vector_path = os.path.join(
+            self.workspace_dir, 'aggregate_vector_3d')
+        vector = driver.CreateDataSource(aggregating_vector_path)
+
+        projection = osr.SpatialReference()
+        projection.ImportFromEPSG(3116)
+        layer = vector.CreateLayer('aggregate', srs=projection,
+                                   geom_type=ogr.wkbMultiPolygonZM)
+        layer_defn = layer.GetLayerDefn()
+
+        origin = (444720, 3751320)
+        polygon_a = [

```

Comment:
I wonder if all these zonal stats tests use the same primitive geometries, and just build them up in different ways to get the different high-order types? If so, do you think it makes sense to define these polygons, and maybe other related variables, someplace where they can be shared? Okay, looking at the rest of the test suite, there are many zonal stats tests and lots of duplicated code, so I won't suggest you do a refactor of all of that here. I might make an issue and do it separately though.